src/main/java/io/cdap/plugin/gcp/publisher/source/GoogleSubscriber.java

```diff
@@ -52,6 +56,17 @@ public void configurePipeline(PipelineConfigurer pipelineConfigurer) {
     FailureCollector collector = pipelineConfigurer.getStageConfigurer().getFailureCollector();
     config.validate(collector);
     pipelineConfigurer.getStageConfigurer().setOutputSchema(config.getSchema());
+    if (!pipelineConfigurer.isFeatureEnabled(Feature.STREAMING_PIPELINE_NATIVE_STATE_TRACKING.getFeatureFlagString())) {
```

Comment:
the feature flag can be on while the pipeline still decides to use Spark checkpointing right? Do we need to expose the checkpointing mode?

Reply:
though I guess it's not wrong to set these when using checkpointing?

Reply:
Hmm, yes, there are a few additional check for using state store like single source, no windower etc. Slightly more involved way would be to add `isRetryable` method in StreamingStateHandler and check and add these in DataStreamsSparkLauncher while launching the pipeline.